src/main/java/org/folio/orders/utils/HelperUtils.java

```diff
@@ -320,37 +320,68 @@ public static String getEndpointWithQuery(String query, Logger logger) {
   }
 
   private static List<Error> validatePoLineWithMixedFormat(CompositePoLine compPOL) {
-    int costPhysicalQuantity = defaultIfNull(compPOL.getCost().getQuantityPhysical(), 0);
-    int costElectronicQuantity = defaultIfNull(compPOL.getCost().getQuantityElectronic(), 0);
-    int locPhysicalQuantity = getPhysicalQuantity(compPOL.getLocations());
-    int locElectronicQuantity = getElectronicQuantity(compPOL.getLocations());
 
     List<ErrorCodes> errors = new ArrayList<>();
     // The quantity of the physical and electronic resources in the cost must be specified
-    if (costPhysicalQuantity + costElectronicQuantity == 0) {
-      errors.add(ErrorCodes.ZERO_COST_QTY);
-    } else if (costPhysicalQuantity == 0) {
+    if (getPhysicalCostQuantity(compPOL) == 0) {
       errors.add(ErrorCodes.ZERO_COST_PHYSICAL_QTY);
-    } else if (costElectronicQuantity == 0) {
-      errors.add(ErrorCodes.ZERO_COST_ELECTRONIC_QTY);
-    }
-    // The total quantity of the physical resources of all locations must not exceed specified in the cost
-    if (locPhysicalQuantity > costPhysicalQuantity) {
-      errors.add(ErrorCodes.PHYSICAL_LOC_QTY_EXCEEDS_COST);
-    } else if (locPhysicalQuantity < costPhysicalQuantity) {
-      errors.add(ErrorCodes.PHYSICAL_COST_QTY_EXCEEDS_LOC);
     }
-    // The total quantity of the electronic resources of all locations must not exceed specified in the cost
-    if (locElectronicQuantity > costElectronicQuantity) {
-      errors.add(ErrorCodes.ELECTRONIC_LOC_QTY_EXCEEDS_COST);
+    if (getElectronicCostQuantity(compPOL) == 0) {
+      errors.add(ErrorCodes.ZERO_COST_ELECTRONIC_QTY);
     }
+    errors.addAll(validateLocations(compPOL));
 
-    validateCostPrices(compPOL.getCost(), P_E_MIX, errors);
+    errors.addAll(validateCostPrices(compPOL));
 
     return convertErrorCodesToErrors(compPOL, errors);
   }
 
-  private static void validateCostPrices(Cost cost, CompositePoLine.OrderFormat orderFormat, List<ErrorCodes> errors) {
+  public static int getPhysicalCostQuantity(CompositePoLine compPOL) {
+    return defaultIfNull(compPOL.getCost().getQuantityPhysical(), 0);
+  }
+
+  public static int getElectronicCostQuantity(CompositePoLine compPOL) {
+    return defaultIfNull(compPOL.getCost().getQuantityElectronic(), 0);
+  }
+
+  private static List<ErrorCodes> validateLocations(CompositePoLine compPOL) {
+    if (isLocationsValidationRequired(compPOL)) {
```

Comment:
Please check `createInventory` separately for Physical and Electronic resources.
In case the PO Line is `P/E Mix` and physical.createInventory is "Instance, Holding, Item" (or "Instance, Holding") but `eresource.createInventory` is "Instance" (or "None"), an user might want to fill in locations info only for physical quantities but validation will force to specify electronic quantities there as well...

Reply:
Done